build/tools/make-winpe-i.MX8.cmd

```diff
@@ -0,0 +1,234 @@
+::
+:: Prepares a WinPE image for boot on i.MX6 or i.MX7
+::
+@echo off
+setlocal enableextensions disabledelayedexpansion
+
+set WINPE_DIR=%ProgramFiles(x86)%\Windows Kits\10\Assessment and Deployment Kit\Windows Preinstallation Environment\arm64
+set DEST=winpe_imx
+set FFU_DISK_NUM=0
+set WINPE_DRIVE_LETTER=D
+set SCRIPT_DIR=%~dp0
+
+:: make WIM mount directory
+rmdir /s /q mount > NUL 2>&1
+mkdir mount
+pushd mount
+set MOUNT_DIR=%CD%
+popd
+
+:: make directory to hold packages
+rmdir /s /q packages > NUL 2>&1
+mkdir packages
+pushd packages
+set PACKAGES_DIR=%CD%
+popd
+
+:: Parse options
+:GETOPTS
+ if /I "%~1" == "/?" goto USAGE
+ if /I "%~1" == "/Help" goto USAGE
+ if /I "%~1" == "/builddir" set BUILD_DIR=%2& shift
+ if /I "%~1" == "/firmware" set FIRMWARE_PATH=%2& shift
+ if /I "%~1" == "/uefi" set UEFI_PATH=%2& shift
+ if /I "%~1" == "/ffu" set FFU_PATH=%2& shift
+ if /I "%~1" == "/ffudisk" set FFU_DISK_NUM=%2& shift
+ if /I "%~1" == "/apply" set DISK_NUM=%2& shift
+ if /I "%~1" == "/clean" set CLEAN=1
+ shift
+if not (%1)==() goto GETOPTS
+
+if not "%CLEAN%" == "" goto CLEAN
+if not "%DISK_NUM%" == "" goto APPLY
+
+if "%BUILD_DIR%" == "" (
+    echo Missing required option '/builddir'. Type /? for usage.
+    exit /b 1
+)
+
+if "%FIRMWARE_PATH%" == "" (
+    echo Missing required option '/firmware'. Type /? for usage.
+    exit /b 1
+)
+
+if "%UEFI_PATH%" == "" (
+    echo Missing required option '/uefi'. Type /? for usage.
+    exit /b 1
+)
+
+:: make temporary WinPE directory
+rmdir /s /q "%DEST%" > NUL 2>&1
+mkdir "%DEST%"
+mkdir "%DEST%\sources"
+
+:: copy firmware files
+echo Copying firmware from %FIRMWARE_PATH%
+copy "%FIRMWARE_PATH%" "%DEST%" /y || goto err
+
+echo Copying UEFI from %UEFI_PATH%
+copy "%UEFI_PATH%" "%DEST%\uefi.fit" /y || goto err
+
+:: copy winpe files
+echo Creating WinPE image layout at %DEST%
+echo Copying WinPE files from %WINPE_DIR%
+if not exist "%WINPE_DIR%\Media" (
+    echo You must install the Windows PE Add-on for the ADK
+    echo https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/download-winpe--windows-pe
+    goto err
+)
+xcopy /herky "%WINPE_DIR%\Media" "%DEST%\" || goto err
+copy "%WINPE_DIR%\en-us\winpe.wim" "%DEST%\sources\boot.wim" || goto err
+move "%DEST%\bootmgr.efi" "%DEST%\EFI\Microsoft\boot\" || goto err
+
+:: BCD
+echo Enabling kernel debugging
+set TARGET_BCD_STORE=%DEST%\EFI\Microsoft\boot\bcd
+bcdedit /store "%TARGET_BCD_STORE%" /dbgsettings SERIAL DEBUGPORT:1 BAUDRATE:115200 || goto err
+bcdedit /store "%TARGET_BCD_STORE%" /debug {default} on || goto err

```

Comment:
Do we want to have KD enabled in the WinPE image by default?

Reply:
I added a new command-line argument to enable it. It's not on by default anymore.